lms/tests/test_models.py

```diff
@@ -1,143 +1,251 @@
+        # Basic functionality
+        assert solution.state == Solution.STATES.CREATED.name
+        marked = solutions.mark_as_checked(solution.id, staff_user.id)
+        # HELL WITH PEEWEE!!!
+        solution = Solution.get_by_id(solution.id)
+        assert marked
+        assert solution.state == Solution.STATES.DONE.name
+        assert solution.checker == staff_user
+
+        # Not duplicating things
+        staff_user2 = conftest.create_staff_user(index=1)
+        solution2 = Solution.create_solution(exercise, student_user)
+        marked = solutions.mark_as_checked(solution2.id, staff_user2.id)
+        solution2 = Solution.get_by_id(solution2.id)
+        assert solution2.state == Solution.STATES.DONE.name
+        assert solution2.checker == staff_user2
+
+        # Creates notifications
+        assert len(list(notifications.get(student_user))) == 2
 
     @staticmethod
-    def assert_password(user, password):
-        assert user.password != password
-        assert password not in user.password
-        assert user.is_password_valid(password)
+    def test_get_next_unchecked(
+            student_user: User,
+            exercise: Exercise,
+            staff_user: User,
+    ):
+        student_user2 = conftest.create_student_user(index=1)
+        exercise2 = conftest.create_exercise(3)
+        solution1 = conftest.create_solution(exercise, student_user)
+        solution2 = conftest.create_solution(exercise2, student_user)
+        solution3 = conftest.create_solution(exercise, student_user2)
 
+        assert len(list(Solution.select())) == 3
 
-class TestSolution:
-    old_solution_state = Solution.STATES.OLD_SOLUTION.name
-    in_checking_state = Solution.STATES.IN_CHECKING.name
-    created_state = Solution.STATES.CREATED.name
+        unchecked = solutions.get_next_unchecked(exercise.id)
+        assert unchecked is not None
+        assert unchecked.exercise.id == solution1.exercise.id
+        assert unchecked == solution1
 
-    def test_new_solution_override_old_solutions(
-            self,
+        solutions.mark_as_checked(solution1.id, staff_user)
+        unchecked = solutions.get_next_unchecked(exercise.id)
+        assert unchecked is not None
+        assert unchecked.exercise.id == solution3.exercise.id
+        assert unchecked == solution3
+
+        solutions.mark_as_checked(solution3.id, staff_user)
+        unchecked = solutions.get_next_unchecked(exercise.id)
+        assert unchecked is None
+
+        unchecked = solutions.get_next_unchecked()
+        assert unchecked is not None
+        assert unchecked == solution2
+
+        solutions.mark_as_checked(solution2.id, staff_user)
+        unchecked = solutions.get_next_unchecked()
+        assert unchecked is None
+
+        unchecked = solutions.get_next_unchecked(solution2.id)
+        assert unchecked is None
+
+    @staticmethod
+    def test_start_checking(
+            db_in_memory: SqliteDatabase,
```

Comment:
why you need this? this fixture is autouse in session scope

Reply:
You are right, leftovers from my flamewar with peewee